github.js

```diff
@@ -584,14 +585,26 @@ GithubLocation.prototype = {
                       mkdirp(path.dirname(fileName), function(err) {
                         if (err)
                           return reject(err);
-                        readStream.pipe(fs.createWriteStream(fileName));
+                        files.push(new Promise(function(_resolve, _reject) {
+                            var p = fs.createWriteStream(fileName).on("close", function(err) {
+                                if (err) _reject(err);
+                                _resolve();
+                            });
+                            readStream.pipe(p);
+                        }));
                       });
                     });
                   });
-
-                  zipFile.on('close', resolve);
+                  zipFile.on('close', function() {
+                      rsvp.all(files).then(function() {
```

Comment:
`Promise.all(files).then(function() { ...` would be the ES promise equivalent here.